Assets/LeapMotionModules/Interaction/Scripts/InteractionController.cs

```diff
@@ -0,0 +1,1382 @@
+      }
+
+      return primaryHoverBegan;
+    }
+
+    /// <summary>
+    /// Called by the Interaction Manager every fixed frame.
+    /// Returns whether any object is the primary hover of this hand and, if so, outputs
+    /// the object into primaryHoveredObject.
+    /// </summary>
+    public bool CheckPrimaryHoverStay(out IInteractionBehaviour primaryHoveredObject) {
+      primaryHoveredObject = _primaryHoveredObject;
+      bool primaryHoverStayed = primaryHoveredObject != null;
+
+      if (primaryHoverStayed && primaryHoveredObject is InteractionBehaviour) {
+        OnStayPrimaryHoveringObject(primaryHoveredObject as InteractionBehaviour);
+      }
+
+      return primaryHoverStayed;
+    }
+
+    #endregion
+
+    /// <summary>
+    /// Returns the hover distance from the hoverPoint to the specified object, automatically
+    /// accounting for ISpaceComponent warping if necessary.
+    /// </summary>
+    public static float GetHoverDistance(Vector3 hoverPoint, IInteractionBehaviour behaviour) {
+      if (behaviour.space != null) {
+        return behaviour.GetHoverDistance(TransformPoint(hoverPoint, behaviour.space));
+      }
+      else {
+        return behaviour.GetHoverDistance(hoverPoint);
+      }
+    }
+
+    /// <summary>
+    /// Applies the spatial warping of the provided ISpaceComponent to a world-space point.
+    /// </summary>
+    public static Vector3 TransformPoint(Vector3 worldPoint, ISpaceComponent element) {
+      if (element.anchor != null && element.anchor.space != null) {
+        Vector3 localPos = element.anchor.space.transform.InverseTransformPoint(worldPoint);
+        return element.anchor.space.transform.TransformPoint(element.anchor.transformer.InverseTransformPoint(localPos));
+      }
+      else {
+        return worldPoint;
+      }
+    }
+
+    #endregion
+
+    #region Contact
+
+    #region Contact Bones
+
+    protected const float DEAD_ZONE_FRACTION = 0.1F;
+
+    public float softContactDislocationFraction = 3.0F;
+
+    private static PhysicMaterial s_defaultContactBoneMaterial;
+    protected static PhysicMaterial defaultContactBoneMaterial {
+      get {
+        if (s_defaultContactBoneMaterial == null) {
+          initDefaultContactBoneMaterial();
+        }
+        return s_defaultContactBoneMaterial;
+      }
+    }
+
+    /// <summary>
+    /// ContactBones should have PhysicMaterials with a bounciness of
+    /// zero and a bounce combine set to minimum.
+    /// </summary>
+    private static void initDefaultContactBoneMaterial() {
+      if (s_defaultContactBoneMaterial == null) {
+        s_defaultContactBoneMaterial = new PhysicMaterial();
+      }
+      s_defaultContactBoneMaterial.hideFlags = HideFlags.HideAndDontSave;
+      s_defaultContactBoneMaterial.bounceCombine = PhysicMaterialCombine.Minimum;
+      s_defaultContactBoneMaterial.bounciness = 0F;
+    }
+
+    private bool _contactInitialized = false;
+    protected abstract ContactBone[] contactBones { get; }
```

Comment:
Anything abstract should be documented so that a user knows what they need to do in order to implement it.